pkg/state/types_test.go

```diff
@@ -0,0 +1,45 @@
+package state_test
+
+import (
+	"encoding/binary"
+	"math/big"
+	"testing"
+
+	"github.com/libp2p/go-libp2p-core/peer"
```

Comment:
I'm striving to avoid depending on libp2p for this entire package. Since the peerid encoding is literally a no-op, depending on this is really unfortunate. Can we just use a manually-constructed byte array?

Reply:
Yeah I can give that a shot

Reply:
Errhh on second thought, this is just for convenience while testing- libp2p-peer isn't used outside of the test file, is that alright?

Reply:
No, that's the whole point. "just for convenience" isn't a good reason to pull in thousands of lines of code. I don't think anything here needs to assert that the byte array forms a valid peer id. it just needs to check that the same bytes come out as went in.

Reply:
alright, addressed it in latest changes